packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -247,6 +252,88 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
+        if (attributeType && visitExistingAttributesTypes.includes(attributeType)) {
+          const attributeToEdit = visitToEdit.attributes.find((attr) => attr.attributeType.uuid === attributeType);
+
+          if (attributeToEdit) {
+            // continue to next attribute if the previous value is same as new value
+            if (typeof attributeToEdit.value === 'object' && attributeToEdit.value.uuid === value) {
+              continue;
+            } else if (attributeToEdit.value === value) {
+              continue;
+            }
+
+            if (value) {
+              // Update attribute with updated value
+              promises.push(
+                openmrsFetch(`/ws/rest/v1/visit/${visitUuid}/attribute/${attributeToEdit.uuid}`, {
+                  method: 'POST',
+                  headers: { 'Content-type': 'application/json' },
+                  body: { value },
+                }).catch((err) => {
+                  showSnackbar({
+                    title: t('errorUpdatingVisitAttribute', 'Could not update {{attributeName}} attribute', {
+                      attributeName: attributeToEdit.attributeType.display,
+                    }),
+                    kind: 'error',
+                    isLowContrast: false,
+                    subtitle: err?.message,
+                  });
+                }),
+              );
+            } else {
+              // Delete attribute if the was no value provided
+              promises.push(
+                openmrsFetch(`/ws/rest/v1/visit/${visitUuid}/attribute/${attributeToEdit.uuid}`, {
+                  method: 'DELETE',
+                }).catch((err) => {
+                  showSnackbar({
+                    title: t('errorDeletingVisitAttribute', 'Could not delete {{attributeName}} attribute', {
+                      attributeName: attributeToEdit.attributeType.display,
+                    }),
+                    kind: 'error',
+                    isLowContrast: false,
+                    subtitle: err?.message,
+                  });
+                }),
+              );
+            }
+          }
+        } else {
+          if (value) {
+            promises.push(
+              openmrsFetch(`/ws/rest/v1/visit/${visitUuid}/attribute`, {
+                method: 'POST',
+                headers: { 'Content-type': 'application/json' },
+                body: { attributeType, value },
+              }).catch((err) => {
+                showSnackbar({
+                  title: t('errorCreatingVisitAttribute', 'Could not delete {{attributeName}} attribute', {
+                    attributeName: visitAttributeTypes?.find((type) => type.uuid === attributeType).display,
+                  }),
+                  kind: 'error',
+                  isLowContrast: false,
+                  subtitle: err?.message,
+                });
+              }),
+            );
+          }
+        }
+      }
```

Comment:
This for loops lacks clarity and I find it hard to read. The primary reason might be those long and tangled if/else blocks. Splitting them into smaller functions or subfunctions would definitely help. Plus, I reckon we could simplify those if/else bits even more.

a rough suggestion:
```js
      if (!attributeType) continue;
      if (sameValue(...) continue;

      const existingAttribute = visitToEdit.attributes.find(
        (attr) => attr.attributeType.uuid === attributeType
      );

      if (existingAttribute) {
        handleExistingAttribute(...);   // or call separate update, delete functions
      } else {
        handleNewAttribute(...);
      }
```

Any thoughts? @usamaidrsk @vasharma05 @ibacher @denniskigen